packages/block-library/src/table/edit.js

```diff
@@ -377,15 +377,24 @@ export class TableEdit extends Component {
 							};
 
 							const cellClasses = classnames( { 'is-selected': isSelected } );
+							const richTextClassName = 'wp-block-table__cell-content';
 
 							return (
 								<CellTag
 									key={ columnIndex }
 									className={ cellClasses }
 									scope={ CellTag === 'th' ? scope : undefined }
+									onClick={ ( event ) => {
+										// When a cell is selected, forward focus to the child RichText. This solves an issue where the
+										// user may click inside a cell, but outside of the RichText, resulting in nothing happening.
+										const richTextElement = event && event.target && event.target.querySelector( `.${ richTextClassName }` );
+										if ( richTextElement ) {
+											richTextElement.focus();
```

Comment:
I removed this line, and the e2e test still passes.

Reply:
Thanks for spotting that, the selector used in the test was wrong, now fixed.